src/main/java/domain/PokemonComparator.kt

```diff
@@ -0,0 +1,32 @@
+package domain
+
+/**
+ * La clase recibe los nombres de los [pokemon] a comparar
+ * Permite comparar 2 pokemon por los siguientes criterios:
+ * ---> byBaseExperience
+ * ---> byTotalStat
+ */
+class PokemonComparator (poke1Name: String, poke2Name: String) {
+    val poke1 = PokedexOffline().getPokemonByName(poke1Name)
+    val poke2 = PokedexOffline().getPokemonByName(poke2Name)
+
+    /**
+     * Compara al [pokemon] según su base de experiencia
+     */
+    fun byBaseExperience(): Pokemon {
+        if (this.poke1.base_experience >= this.poke2.base_experience)
+            return poke1
```

Comment:
Te conviene no ser inconsistente y acá poner `this.poke1`. Item en la línea de abajo. Siempre usar el `this.` cuando te referís a variables de instancia, igual que como hacías en javascript.